### PR TITLE
feat: update flagd provider to use float api

### PR DESCRIPTION
**Pull request by beeme1mr**
Signed-off-by: Michael Beemer <michael.beemer@dynatrace.com>

### CODE REVIEW

File: libs/providers/flagd/src/lib/flagd-provider.spec.ts [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
 
   describe('http service tests', () => {
     const axiosMock = new MockAdapter(axios);
-    let client: Client
+    let client: Client;
 
     beforeEach(() => {
       OpenFeature.setProvider(new FlagdProvider());
-      client = OpenFeature.getClient("test")
+      client = OpenFeature.getClient('test');
```
Comment (toddbaert) [inline libs/providers/flagd/src/lib/flagd-provider.spec.ts:24]: Do we need linting in this repo?

File: libs/providers/flagd/src/lib/service/http/service.ts [hunk 2]
```diff
@@ -37,20 +46,24 @@ export class HTTPService implements Service {
...
+  async resolveNumber(
+    flagKey: string,
+    defaultValue: number,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<number>> {
     try {
-      const res = await axios.post<ResolveNumberResponse>(
-        `${this.url}/flags/${encodeURIComponent(flagKey)}/resolve/number`,
+      const res = await axios.post<ResolveFloatResponse>(
+        `${this.url}/flags/${encodeURIComponent(flagKey)}/resolve/float`,
```
Comment (toddbaert) [inline libs/providers/flagd/src/lib/service/http/service.ts:66]: I think maybe this deserves a comment.

**Review by toddbaert — APPROVED** [review]

File: libs/providers/flagd/src/lib/service/http/service.ts (hunk 2 above)
Comment (beeme1mr) [reply libs/providers/flagd/src/lib/service/http/service.ts:69]: Done

File: libs/providers/flagd/src/lib/flagd-provider.spec.ts (hunk 1 above)
Comment (beeme1mr) [reply libs/providers/flagd/src/lib/flagd-provider.spec.ts:24]: https://github.com/open-feature/node-sdk-contrib/issues/88

**Pull request merged**